catroid/src/main/java/org/catrobat/catroid/stage/StageActivity.java

```diff
@@ -142,13 +150,25 @@ public void onCreate(Bundle savedInstanceState) {
 	}
 
 	private void setupAskHandler() {
+		final StageActivity currentStage = this;
 		messageHandler = new Handler(Looper.getMainLooper()) {
 			@Override
 			public void handleMessage(Message message) {
 				ArrayList<Object> params = (ArrayList<Object>) message.obj;
 
-				if (message.what == ASK_MESSAGE) {
-					showDialog((String) params.get(1), (AskAction) params.get(0));
+				switch (message.what)
+				{
+					case ASK_MESSAGE:
+						showDialog((String) params.get(1), (AskAction) params.get(0));
+						break;
+					case REGISTER_INTENT:
+						currentStage.queueIntent((IntentListener) params.get(0));
+						break;
+					case PERFORM_INTENT:
+						currentStage.startQueuedIntent((Integer) params.get(0));
+						break;
+					default:
+						Log.e(TAG,"Unhandled message in messagehandler, case "+message.what);
```

Comment:
eventhough not necessary, please add a break also at the default case.. this prevents errors if these case statements are reordered